main.go

```diff
@@ -117,9 +119,10 @@ func main() {
 	}()
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
-		Scheme:    scheme,
-		Port:      0, // Add flag with default of 9443 when adding webhooks
-		Namespace: watchNamespace,
+		Scheme:                scheme,
+		Port:                  0, // Add flag with default of 9443 when adding webhooks
+		Namespace:             watchNamespace,
+		ClientDisableCacheFor: []client.Object{&corev1.Secret{}},
```

Comment:
We definitely don't want to disable the cache I don't think, just limit it to Secrets we care about with:

```
NewCache: cache.BuilderWithOptions(cache.Options{
	SelectorsByObject: cache.SelectorsByObject{
		&corev1.Secret{}: {
			Label: labels.SelectorFromSet(
				labels.Set{
					"environment.metal3.io": "baremetal"
				}),
		},
	}
}),
```

Bonus points for using the library function as in https://github.com/metal3-io/baremetal-operator/blob/master/main.go#L135-L137

This won't make much difference today, but if one day we are running without namespacing then limiting it to labelled Secrets will save a lot of memory.